var/spack/repos/builtin/packages/babelstream/package.py

```diff
@@ -140,38 +158,67 @@ class Babelstream(CMakePackage, CudaPackage, ROCmPackage):
     variant("flags", values=str, default="none", description="Additional CXX flags to be provided")
 
     # comp_impl_vals=["ONEAPI-DPCPP","DPCPP","HIPSYCL","COMPUTECPP"]
-    variant(
-        "implementation",
-        values=str,
-        default="none",
-        description="Compile using the specified SYCL compiler option",
-    )
-
-    conflicts(
-        "implementation=none",
-        when="+sycl",
-        msg="SYCL requires compiler implementation to be specified by option=",
-    )
-    conflicts(
-        "implementation=none",
-        when="+thrust",
-        msg="Which Thrust implementation to use, supported options include:\
-         - CUDA (via https://github.com/NVIDIA/thrust)\
-         - ROCM (via https://github.com/ROCmSoftwarePlatform/rocThrust)",
-    )
+    with when("build_system=cmake"):
+        variant(
+            "implementation",
```

Comment:
This name seems less specific than ideal

Reply:
Since some descriptions were referred to as `implementation` in the original CMake package, having a similar naming trend could be easier to follow. However, brainstorming a new variant name is an option.